### ISSUE TITLE
Frozen String Literal Behavior is Confusing

### ISSUE THREAD

**Issue by gfyoung** [rubocop-hq/rubocop#7426]
According to <a href="https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/FrozenStringLiteralComment">the docs</a>, my understanding was that in order for the cop to be satisfied, we had to add `# frozen_string_literal: true` to the top of the file.

However, after finding what appeared to be a false negative in one of my projects, I found #7241 and its corresponding fix in #7251 that highlighted that adding `# frozen_string_literal: false` to the top can still pass this cop.  Two points:

* I think the docs should be updated to reflect that `false` is accepted in the magic comment
* Is there a way to configure the cop so that it enforces `# frozen_string_literal: true` only?

## Rubocop Info

~~~
0.75.0 (using Parser 2.6.5.0, running on ruby 2.6.5 x86_64-linux)
~~~

**Comment by koic** [comment]
Good points.

> I think the docs should be updated to reflect that `false` is accepted in the magic comment

Yeah, it will be easy to understand if it is documented.

> Is there a way to configure the cop so that it enforces `# frozen_string_literal: true` only?

AFAIK, there is no such configure option, so enhancement will be necessary.

**Comment by gfyoung** [comment]
@koic : Thanks for your reply!

I imagine a docs change is something people wouldn't object to.

The enhancement would be nice.  Would be curious to know if people have any objections.

**Comment by olliebennett** [comment]
I assumed the `Style/FrozenStringLiteralComment` cop already checked for `true`, to be honest.

For reference, current RuboCop [default configuration](https://github.com/rubocop-hq/rubocop/blob/master/config/default.yml#L2731) options are;

```
  EnforcedStyle: always
  SupportedStyles:
    # `always` will always add the frozen string literal comment to a file
    # regardless of the Ruby version or if `freeze` or `<<` are called on a
    # string literal. If you run code against multiple versions of Ruby, it is
    # possible that this will create errors in Ruby 2.3.0+.
    - always
    # `never` will enforce that the frozen string literal comment does not
    # exist in a file.
    - never
```

Would it make sense to add a new SupportedStyle of `alwaystrue`?

```
    # `alwaystrue` will enforce that the frozen string literal comment exists
    # and is set to `true` (i.e. enforcing that string literals _are_ frozen). 
    - alwaystrue
```

I think defaulting to `always` (the current default) is most intuitive. If someone explicitly uses `frozen_string_literal: false` they are _intentionally_ not freezing strings, rather than _accidentally_ not freezing strings. An `alwaystrue` configuration option hopefully solves everyone's use-cases, while maintaining backwards compatibility.

**Comment by gfyoung** [comment]
@olliebennett : My thoughts exactly.  I wanted to see if there were any maintainer objections though on this before proceeding with an enhancement.
